### PR TITLE
Adding iPXE efi binary build to image builder

### PR DESCRIPTION
**Pull request by keedya**

### CODE REVIEW

**Comment by heckj** [comment]
@keedya I think I screwed you on a earlier merge - this PR needs a rebase, but also thought you might have an edit on roles/ipxe/files/general.h to remove the COMBOOT option

File: roles/ipxe/tasks/main.yml [hunk 1]
```diff
@@ -12,6 +12,11 @@
   apt: name=genisoimage state=present
   sudo: yes
 
+- name: install binutils-dev libiberty-dev
+  apt: name=binutils-dev state=present
+  apt: name=libiberty-dev state=present
```
Comment (benbp) [inline roles/ipxe/tasks/main.yml:20]: Can you do this with ansible, two `apt` objects in the same task? Either way, probably better written as:

```
- name: install binutils-dev and libiberty-dev
  apt: name={{ item }} state=present
  sudo: yes
  with_items:
    - binutils-dev
    - libiberty-dev
```

**Comment by keedya** [comment]
@heckj @benbp @jlongever check it now

**Comment by heckj** [comment]
@keedya I don't see any change to the COMBOOT setting in the files, is that going to screw you for the UEFI build?

**Comment by keedya** [comment]
@heckj this is not a issue for 64 bit binary.

File: roles/ipxe/tasks/main.yml [hunk 2]
```diff
@@ -48,6 +62,20 @@
     - ipxe.pxe
     - undionly.kpxe
 
+- name: fetch the resulting files efi 64 bit file
+  fetch: src={{ ansible_env.HOME }}/ipxe/src/bin-x86_64-efi/{{ item }}
+        dest=/tmp/on-imagebuilder/ipxe/bin-x86_64-efi{{ item }} fail_on_missing=yes
```
Comment (benbp) [inline roles/ipxe/tasks/main.yml]: Can we set this directory path as a variable instead? Perhaps in https://github.com/RackHD/on-imagebuilder/blob/master/hosts#L21 or in group_vars?

**Comment by benbp** [comment]
Failing https://travis-ci.org/RackHD/on-imagebuilder/builds/91014153#L581

**Comment by keedya** [comment]
failed because of the following libiberty-dev

File: roles/ipxe/tasks/main.yml [hunk 3]
```diff
@@ -40,6 +44,16 @@
   args:
     chdir: "{{ ansible_env.HOME }}/ipxe/src"
 
+- name: build the general efi 64
+  command: make bin-x86_64-efi/snponly.efi EMBED=default.ipxe
+  args:
+    chdir: "{{ ansible_env.HOME }}/ipxe/src"
+
+#- name: build the general code efi 32
```
Comment (jlongever) [inline roles/ipxe/tasks/main.yml]: Why are we skipping efi32 images?

Comment (jlongever) [inline roles/ipxe/tasks/main.yml]: I thought we were going to remove COMBOOT and leave efi32 build?

**Comment by benbp** [comment]
Same failure as before.

**Comment by benbp** [comment]
:+1: though I'll mention your latest 3 commits aren't authored by your git user (onrack instead).

**Comment by jlongever** [comment]
:+1:

**Pull request merged**